### ISSUE TITLE
Search in new version throws exception

### ISSUE THREAD

**Issue by bjoern-vh** [lekma/plugin.video.invidious#103]
Hello lekma,

thanks for your new release. Searching is not working for me. I am using Kodi in newest version 21.1 (LibreELEC 12.1) on a Raspberry Pi 4B with 4 GB RAM. I am running my own Invidious instance (Version: 2024.09.20-a021b93) and it's working well in browser. All functions are available and working.

I get this error when I click on "Search":
`
error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'KeyError'>
                                                   Error Contents: 'upload_date'
                                                   Traceback (most recent call last):
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 221, in <module>
                                                       dispatch(*argv)
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 217, in dispatch
                                                       IVPlugin(url, int(handle)).dispatch(**parseQuery(query))
                                                     File "/storage/.kodi/addons/script.module.nuttig/lib/nuttig/plugin.py", line 72, in dispatch
                                                       return action(**kwargs)
                                                              ^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/script.module.nuttig/lib/nuttig/plugin.py", line 37, in wrapper
                                                       raise error
                                                     File "/storage/.kodi/addons/script.module.nuttig/lib/nuttig/plugin.py", line 34, in wrapper
                                                       success = func(self, **kwargs)
                                                                 ^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 204, in search
                                                       return self.__history__()
                                                              ^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 190, in __history__
                                                       return self.addDirectory(self.__client__.history())
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 43, in addDirectory
                                                       if super(IVPlugin, self).addDirectory(items, *args):
                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/script.module.nuttig/lib/nuttig/plugin.py", line 104, in addDirectory
                                                       return self.addItems(items, *args, **kwargs)
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/script.module.nuttig/lib/nuttig/plugin.py", line 89, in addItems
                                                       self.__handle__, [item.asItem() for item in items if item]
                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/script.module.nuttig/lib/nuttig/plugin.py", line 89, in <listcomp>
                                                       self.__handle__, [item.asItem() for item in items if item]
                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/script.module.nuttig/lib/nuttig/objects.py", line 85, in <genexpr>
                                                       return (item.getItem(*args, **kwargs) for item in self if item)
                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/invidious/items.py", line 158, in getItem
                                                       _sort_=localizedString(querySort[self.sort])
                                                                              ~~~~~~~~~^^^^^^^^^^^
                                                   KeyError: 'upload_date'
                                                   -->End of Python script error report<--

` 
I hope it's only a small thing and you can fix it easily :-)

**Comment by lekma** [comment]
something is wrong here
there is no 'upload_date' in the code (none that i can find at least)
your best bet is removing the profile dir for invidious (.../userdata/addon_data/plugin.video.invidious) and restart.
but ,out of curiosity, would you post your searchhistory.json?

**Comment by lekma** [comment]
i think i know what's happening here, my mistake, let me have a stab at fixing it

**Comment by lekma** [comment]
can you test master?

**Comment by lekma** [comment]
also, sorry for the mistake

**Comment by bjoern-vh** [comment]
@lekma: As always: Thanks for your quick help :-)

I have downloaded master.zip and updated it but even after restarting Kodi the exception persists.

After that I have renamed my old userdata folder for this plugin and restarted again. This works. 

The search history was already deleted except one entry: [test]
Now the new one is: {"test": {"q": "test", "type": "all", "sort": "relevance", "page": 1}}

I don't know where the [test] comes from because I never searched before. The search in new version has not working before and my old search contains other words. The new one is from me for testing purposes.

And now I could copy the old search history and even this works. 

At the end I tried to install the "old" 3.2.0 version from repo and it's working, too!? 
I don't know what went wrong but now all works well with my old data :-)
